module/VuFind/src/VuFind/Controller/ApiControllerTrait.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * API Controller Trait
+ *
+ * PHP version 5
+ *
+ * Copyright (C) The National Library of Finland 2016.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+namespace VuFind\Controller;
+
+use VuFind\Exception\Forbidden as ForbiddenException;
+use Zend\Mvc\MvcEvent;
+
+/**
+ * API Controller Trait
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+trait ApiControllerTrait
```

Comment:
Yes, that's better. I modeled it along the lines of #608 and if that gets in first, I can adapt this (I think it should be promptLogin instead of promptlogin if showMessage is also camelCase).